src/test/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -782,6 +783,57 @@ public void bulkIngestRandomVectors(String indexName, String fieldName, int numV
 
     }
 
+    public void bulkAddKnnDocs(String index, String fieldName,  float[][] indexVectors, int docCount) throws IOException {
+        Request request = new Request(
+                "POST",
+                "/_bulk"
+        );
+
+        request.addParameter("refresh", "true");
+        StringBuilder sb = new StringBuilder();
+
+        for(int i = 0; i < docCount; i++){
+            sb.append("{ \"index\" : { \"_index\" : \"" + index + "\", \"_id\" : \"" + String.valueOf(i+1) + "\" } }\n"
+                    +"{ \"" + fieldName + "\" : "+ Arrays.toString(indexVectors[i]) +" }\n");
```

Comment:
to utilize StringBuilder features strings need to be added via append() method, something like 
sb.append("{ \"index\" : { \"_index\" : \"")
     .append(index)
     .append("\", \"_id\" : \"")
     .append(i + 1)
.....
otherwise each concatenation with "+" creates new String object

Reply:
Got it, updated